bst_plugin/forward/process_nst_import_head_model.m

```diff
@@ -98,224 +96,275 @@
-sCortex = in_tess_bst(sSubject.Surface(sSubject.iCortex ).FileName);
+OPTIONS = struct();
+OPTIONS.SubjectName     = sInputs.SubjectName;
+OPTIONS.MriFile         = sSubject.Anatomy(sSubject.iAnatomy).FileName;
+OPTIONS.HeadFile        = sSubject.Surface(sSubject.iScalp  ).FileName;
+OPTIONS.CortexFile      = sSubject.Surface(sSubject.iCortex ).FileName;
+OPTIONS.FluenceFolder   = sProcess.options.data_source.Value;
+OPTIONS.ChannelFile     = sInputs(1).ChannelFile;
 
-% Retrieve optode coordinates
-% Load channel file
+OPTIONS.smoothing_method    = sProcess.options.method.Value;
+OPTIONS.smoothing_fwhm      = sProcess.options.smoothing_fwhm.Value{1};
 
-montage_info    = nst_montage_info_from_bst_channels(ChannelMat.Channel);
-pair_names      = montage_info.pair_names;
-src_locs        = montage_info.src_pos;
-src_ids         = montage_info.src_ids;
-det_locs        = montage_info.det_pos;
-det_ids         = montage_info.det_ids;
-pair_sd_idx     = montage_info.pair_sd_indexes;
+% Compute Head model
+HeadModelMat = Compute(OPTIONS);
 
-nb_wavelengths = length(ChannelMat.Nirs.Wavelengths);
-nb_sources     = size(src_locs, 1);
-nb_dets        = size(det_locs, 1);
-nb_pairs       = length(pair_names);
 
+% Save Head Model
+sStudy = bst_get('Study', sInputs.iStudy);
 
-% Find closest head vertices (for which we have fluence data)
-% Put everything in mri referential
+HeadModelFile = bst_fullfile(bst_fileparts(file_fullpath(sStudy.FileName)), 'headmodel_nirs_mcx_fluence.mat');
+HeadModelFile = file_unique(HeadModelFile);
+bst_save(HeadModelFile, HeadModelMat, 'v7');
 
-[src_hvidx, det_hvidx] = get_head_vertices_closest_to_optodes(sMri, sHead, src_locs, det_locs);
+% Update Study structure
+newHeadModel = db_template('HeadModel');
+newHeadModel.FileName = file_short(HeadModelFile);
+newHeadModel.Comment = 'NIRS head model from fluence';
+newHeadModel.HeadModelType  = 'surface';    
 
-%% Load fluence data from local .brainstorm folder (download if not available)
-use_closest_wl = 0;
-[all_fluences_flat_sparse, all_reference_voxels_index]= request_fluences([src_hvidx ; det_hvidx], sMri.Comment, ...
-                                                                         ChannelMat.Nirs.Wavelengths, data_source, nan, nan, [], '',...
-                                                                         use_closest_wl);
-if isempty(all_fluences_flat_sparse)
-    return;
+iHeadModel = length(sStudy.HeadModel) + 1;
+if ~isempty(sStudy.HeadModel)
+    sStudy.HeadModel(end+1) = newHeadModel(1);
+else
+    sStudy.HeadModel = newHeadModel;
 end
+sStudy.iHeadModel = iHeadModel;
 
-bst_progress('start', 'Export fluences','Unpacking volumic fluences...', 1, (nb_sources+nb_dets)*nb_wavelengths);
-
-mri_zeros       = zeros(size(sMri.Cube));
-src_fluences    = cell(1, nb_sources);
-det_fluences    = cell(1, nb_dets);
+% Update DataBase
+bst_set('Study', sInputs.iStudy, sStudy);
+panel_protocols('UpdateNode', 'Study', sInputs.iStudy);
+db_save();
```

Comment:
@rcassani can this part of the code be simplified when saving the head model in the databse ?